[BUG]: `_has_cuGraphNodeGetParams` check missed the binding version check

### Is this a duplicate?

- [x] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cuda-python/issues) for this bug and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Type of Bug

Runtime Error

### Component

cuda.core

### Describe the bug

```
____________________________________________________________ test_reconstructed_body_survives_parent_deletion ____________________________________________________________

init_cuda = <Device 0 (NVIDIA RTX 6000 Ada Generation)>

    def test_reconstructed_body_survives_parent_deletion(init_cuda):
        """Body graph obtained via nodes() reconstruction survives parent deletion."""
        g = GraphDefinition()
        condition = try_create_condition(g)
        g.while_loop(condition)
    
>       all_nodes = g.nodes()
                    ^^^^^^^^^

condition  = <GraphCondition handle=0x1>
g          = <GraphDefinition handle=0x5c76568fe9c0>
init_cuda  = <Device 0 (NVIDIA RTX 6000 Ada Generation)>

tests/graph/test_graph_definition_lifetime.py:120: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
cuda/core/graph/_graph_definition.pyx:328: in cuda.core.graph._graph_definition.GraphDefinition.nodes
    return {GraphNode._create(self._h_graph, nodes_vec[i]) for i in range(num_nodes)}
cuda/core/graph/_graph_node.pyx:90: in cuda.core.graph._graph_node.GraphNode._create
    return GN_create(h_graph, node)
cuda/core/graph/_graph_node.pyx:581: in cuda.core.graph._graph_node.GN_create
    return _registered(GN_create_impl(h_node))
cuda/core/graph/_graph_node.pyx:610: in cuda.core.graph._graph_node.GN_create_impl
    return ConditionalNode._create_from_driver(h_node)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   params = handle_return(driver.cuGraphNodeGetParams(
E   AttributeError: module 'cuda.bindings.driver' has no attribute 'cuGraphNodeGetParams'


cuda/core/graph/_subclasses.pyx:641: AttributeError
======================================================================== short test summary info =========================================================================
FAILED tests/graph/test_graph_definition_lifetime.py::test_reconstructed_body_survives_parent_deletion - AttributeError: module 'cuda.bindings.driver' has no attribute 'cuGraphNodeGetParams'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=============================================================== 1 failed, 252 passed, 35 skipped in 6.74s ================================================================
```

### How to Reproduce

Use cuda-bindings 13.1 + CUDA driver 13.2 + latest cuda-core

### Expected behavior

test pass

### Operating System

_No response_

### nvidia-smi output

_No response_

--- a/cuda_core/cuda/core/_module.pyx
+++ b/cuda_core/cuda/core/_module.pyx
@@ -32,7 +32,7 @@ from cuda.core._utils.clear_error_support import (
     raise_code_path_meant_to_be_unreachable,
 )
 from cuda.core._utils.cuda_utils cimport HANDLE_RETURN
-from cuda.core._utils.version cimport cy_driver_version
+from cuda.core._utils.version cimport cy_binding_version, cy_driver_version
 from cuda.core._utils.cuda_utils import driver
 from cuda.bindings cimport cydriver
 
@@ -462,6 +462,11 @@ cdef class Kernel:
             raise NotImplementedError(
                 "Driver version 12.4 or newer is required for this function. "
                 f"Using driver version {'.'.join(map(str, cy_driver_version()))}"
+            )
+        if cy_binding_version() < (12, 4, 0):
+            raise NotImplementedError(
+                "cuda.bindings 12.4 or newer is required for this function. "
+                f"Using binding version {'.'.join(map(str, cy_binding_version()))}"
             )
         cdef size_t arg_pos = 0
         cdef list param_info_data = []

--- a/cuda_core/cuda/core/graph/_subclasses.pyx
+++ b/cuda_core/cuda/core/graph/_subclasses.pyx
@@ -60,8 +60,10 @@ cdef bint _version_checked = False
 cdef bint _check_node_get_params():
     global _has_cuGraphNodeGetParams, _version_checked
     if not _version_checked:
-        from cuda.core._utils.version import driver_version
-        _has_cuGraphNodeGetParams = driver_version() >= (13, 2, 0)
+        from cuda.core._utils.version import binding_version, driver_version
+        _has_cuGraphNodeGetParams = (
+            driver_version() >= (13, 2, 0) and binding_version() >= (13, 2, 0)
+        )
         _version_checked = True
     return _has_cuGraphNodeGetParams
 

--- a/cuda_core/tests/graph/test_graph_definition.py
+++ b/cuda_core/tests/graph/test_graph_definition.py
@@ -48,13 +48,13 @@ def _skip_if_no_managed_mempool():
         pytest.skip("Device does not support managed memory pool operations")
 
 
-def _driver_has_node_get_params():
-    from cuda.core._utils.version import driver_version
+def _has_node_get_params():
+    from cuda.core._utils.version import binding_version, driver_version
 
-    return driver_version() >= (13, 2, 0)
+    return driver_version() >= (13, 2, 0) and binding_version() >= (13, 2, 0)
 
 
-_HAS_NODE_GET_PARAMS = _driver_has_node_get_params()
+_HAS_NODE_GET_PARAMS = _has_node_get_params()
 
 
 def _bindings_major_version():

